src/view/pane/message.js

```diff
@@ -110,8 +110,20 @@ Candy.View.Pane = (function(self, $) {
         timestamp = Candy.Util.iso8601toDate(timestamp);
       }
 
+      var messagePane = self.Room.getPane(roomJid, ".message-pane");
```

Comment:
I think the best we could do here is to make this behaviour optional and enable it when all clients are controlled or meet the requirements for globally unique IDs.